### ISSUE TITLE
Remove type of service in service host constructors

### ISSUE THREAD

**Issue by luboshl** [tonysneed/WCF.CommonInstanceFactory#2]
When I look at samples or tests, there is for example:

``` C#
new SimpleInjectorServiceHost<GreetingService>(_container, typeof(GreetingService))
```

The second argument in the constructor (`typeof(GreetingService)`) is redundant. It is givven by the generic type `<GreetingService>`. So it can be simplified to

``` C#
new SimpleInjectorServiceHost<GreetingService>(_container)
```

I suggest to add new constructor for `SimpleInjectorServiceHost` which call the original construtor:

``` C#
public SimpleInjectorServiceHost(Container container, params Uri[] baseAddresses)
    : this(container, typeof(TServiceType), baseAddresses)
{
}
```

Of course the same with `NinjectServiceHost`.

The original constructor can be marketd as `[Obsolete]`.

Or the original one can be replaced, but this is incompatible change.

**Comment by tonysneed** [comment]
I have a memory that this parameter was needed for some reason. Try submitting a PR with the new ctor overload and leave the existing ctor in place for now. Thanks!

**Comment by luboshl** [comment]
I'll do it after the PR #1 will be accepted, will I?

**Comment by luboshl** [comment]
PR #3 is ready.

**Comment by tonysneed** [comment]
Thanks!

**Comment by tonysneed** [comment]
Resolved by PR #5.
